android/src/main/kotlin/dev/rexios/wear_plus/WearPlugin.kt

```diff
@@ -158,4 +133,15 @@ class WearPlugin : FlutterPlugin, ActivityAware, MethodCallHandler, LifecycleObs
             mMethodChannel?.invokeMethod("onInvalidateAmbientOffload", null)
         }
     }
+
+    override fun onStateChanged(source: LifecycleOwner, event: Lifecycle.Event) {
+        when (event) {
+            Lifecycle.Event.ON_CREATE -> mAmbientController?.onCreate()
+            Lifecycle.Event.ON_RESUME -> mAmbientController?.onResume()
+            Lifecycle.Event.ON_PAUSE -> mAmbientController?.onPause()
+            Lifecycle.Event.ON_STOP -> mAmbientController?.onStop()
+            Lifecycle.Event.ON_DESTROY -> mAmbientController?.onDestroy()
+            else -> {}
```

Comment:
This was the only part where I have some doubts if leaving it empty or just throw an exception since the WearableActivityController don't have those extra lifecycles. @Rexios80

Reply:
We should definitely _not_ throw here because the controller only cares about those events. Other events happening doesn't mean something is wrong.